src/core/analysis/Model.cpp

```diff
@@ -321,13 +318,16 @@ void Model::getOrderedStochasticNodes(const DagNode* dagNode, std::vector<DagNod
     }
 
     // Then I can add myself to the nodes visited, and to the ordered vector of stochastic nodes
-    if ( dagNode->isStochastic() ) //if the node is stochastic
+    if ( the_dag_node->isStochastic() && (visitedNodes.find(the_dag_node) == visitedNodes.end()) ) //if the node is stochastic
```

Comment:
Wow, cool change!  I wonder if this fixes the problem with:
```
for (i in 1:98) {

  dx[i] ~ dnNormal(0,1)

  x[i+2] := dx[i] - x[i] + 2 * x[i+1]

  # # Use below print statement to watch how time to compute probability slows as model is built

  print(x[1].lnProbability())
}
```

Reply:
No, that still seems to be a problem.